app/celery/scheduled_tasks.py

```diff
@@ -153,25 +154,36 @@ def _check_slow_text_message_delivery_reports_and_raise_error_if_needed(reports:
     # sentry error for us to investigate.
     if percent_slow_notifications >= 10:
-        with sentry_sdk.push_scope() as scope:
-            error_context = {
-                "Support runbook": (
-                    "https://github.com/alphagov/notifications-manuals/wiki/Support-Runbook#slow-sms-delivery"
-                ),
-                "Slow text messages - #": slow_notifications,
-                "Slow text messages - %": percent_slow_notifications,
-                "Total text messages": total_notifications,
-            }
+        count = redis_store.incr(CacheKeys.NUMBER_OF_TIMES_OVER_SLOW_SMS_DELIVERY_THRESHOLD)
+
+        # If this is the third consecutive time we've seen the threshold breached, then we log an error to Sentry.
+        # This tells us that for at least 15 minutes we've seen delivery take longer than 5 minutes for >10% of
+        # requests. At that point it's probably worth investigating. By only checking for == 3, we don't log any
+        # further errors until we recover and then starting slowing down again. This should mean that each instance
+        # of the error on Sentry actually deserves to be investigated as a separate issue/potential incident.
+        if count == 3:
+            with sentry_sdk.push_scope() as scope:
+                error_context = {
+                    "Support runbook": (
+                        "https://github.com/alphagov/notifications-manuals/wiki/Support-Runbook#slow-sms-delivery"
+                    ),
+                    "Slow text messages - #": slow_notifications,
+                    "Slow text messages - %": percent_slow_notifications,
+                    "Total text messages": total_notifications,
+                }
 
-            for report in reports:
-                error_context[f"provider.{report.provider}.slow_ratio"] = report.slow_ratio
-                error_context[f"provider.{report.provider}.slow_notifications"] = report.slow_notifications
-                error_context[f"provider.{report.provider}.total_notifications"] = report.total_notifications
+                for report in reports:
+                    error_context[f"provider.{report.provider}.slow_ratio"] = report.slow_ratio
+                    error_context[f"provider.{report.provider}.slow_notifications"] = report.slow_notifications
+                    error_context[f"provider.{report.provider}.total_notifications"] = report.total_notifications
 
-            scope.set_context("Slow SMS delivery", error_context)
-            current_app.logger.error(
-                "Over 10% of text messages sent in the last 15 minutes have taken over 5 minutes to deliver."
-            )
+                scope.set_context("Slow SMS delivery", error_context)
+                current_app.logger.error(
+                    "Over 10% of text messages sent in the last 15 minutes have taken over 5 minutes to deliver."
```

Comment:
Should we update this wording? Am I right in thinking it is the last 45 minutes 10% of the text messages have taken over 5 minutes to deliver?

Reply:
Not quite. Exactly what this means is a bit tricky to pin down.

However this task runs every 5 minutes, so 3 consecutive failures will take 10 minutes to happen (first, then 5 mins later, then another 5 mins later).

Exactly what this necessarily means is a bit hard for me to pin down mentally even in my own head. There's quite a few ways we could hit this issue...

We could have consistent delivery issues where 10% of _all_ messages are taking >5 mins to deliver. We could have a little burst where, in some 5 minute period, 30% of _all_ messages get delayed for >5 mins and then nothing else is delayed at all. That 5 minute period will show up in 3 of these checks, and averaged over 15 minutes will show up as 10% of all messages in a 15 minute period. etc

Reply:
Actually ... this task runs every minute, which I didn't realise 🙃  Which makes a lot of my above 'logic' incorrect.

I might change this to require 5 consecutive detections.

Reply:
Makes sense. I think the error message should be updated to be truthful. So if it looks back 15 minutes, and runs every minute then for 5 runs it would be 19 minutes? Does that make sense? Or could we just extend the window we look back and not have it have to have multiple failures?

Reply:
I want to keep the message fairly succinct and I believe the current error message _is_ truthful, it's just not _exacting_.

Every time we record a failure, it must be the case that within the last 15 minutes, 10% or more of text messages have taken 5 minutes to deliver.

We just only actually error on this if that is the case 5x in a row, so as to provide us some certainty that it's not just a blip in delivery but a persistent issue we need to investigate.

---

I'm open to alternative suggestions but I don't want to get too blocked on this as we can always change the message in the future.